user_service_basic.go

```diff
@@ -13,22 +13,22 @@ import (
 // BasicUserManager implements the UserManager interface and has a list of
 // BasicUsers which is passed into the constructor.
 type BasicUserManager struct {
-	users []basicUser
+	users []*basicUser
```

Comment:
is it possible to avoid the indirection here.

Reply:
Yeah, but it wouldn't be as elegant.